lib/CL/pocl_llvm_api.cc

```diff
@@ -262,22 +202,17 @@ int pocl_llvm_build_program(cl_program program,
   CompilerInstance CI;
   CompilerInvocation &pocl_build = CI.getInvocation();
 
-  // add device specific switches, if any
   std::stringstream ss;
   std::stringstream ss_build_log;
 
-  std::stringstream build_log_filename;
-  build_log_filename << cache_dir << "/" << POCL_BUILDLOG_FILENAME;
-  /* Overwrite build log */
-  std::ofstream fp(build_log_filename.str().c_str(), std::ofstream::trunc);
-  fp.close();
-
-  if (device->ops->init_build != NULL) 
+  // add device specific switches, if any
+  // TODO this currently passes NULL as device tmpdir
+  cl_device_id device = program->devices[device_i];
+  if (device->ops->init_build != NULL)
     {
-      assert (device_tmpdir != NULL);
-      char *device_switches = 
-        device->ops->init_build (device->data, device_tmpdir);
-      if (device_switches != NULL) 
+      char *device_switches =
+        device->ops->init_build (device->data, NULL);
```

Comment:
Better remove the device_tmpdir argument from the driver API if it's not supported.